src/neighbors.cc

```diff
@@ -1148,22 +1149,33 @@ peerDNSConfigure(const ipcache_addrs *ia, const Dns::LookupDetails &, void *data
 }
 
 static void
-peerRefreshDNS(void *data)
+peerDnsRefreshStart()
 {
-    if (eventFind(peerRefreshDNS, nullptr))
-        eventDelete(peerRefreshDNS, nullptr);
+    for (const auto &p: CurrentCachePeers())
+        ipcache_nbgethostbyname(p->host, peerDNSConfigure, p.get());
 
+    /* Reconfigure the peers every hour */
+    peerScheduleDnsRefreshCheck(3600.0);
+}
+
+static void
+peerDnsRefreshCheck(void *data)
+{
     if (!data && 0 == stat5minClientRequests()) {
```

Comment:
I assume you are working on removing `data` from this condition and will come back to this PR after that is done.

Reply:
Done.